<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Generate and visualise synthetic Whole Genome Sequence data • generatervis</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Generate and visualise synthetic Whole Genome Sequence data">
<meta name="description" content="Generate and visualise synthetic Whole Genome Sequence data This package provides a research data management workflow to create synthetic Whole Genome Sequence clinical data.">
<meta property="og:description" content="Generate and visualise synthetic Whole Genome Sequence data This package provides a research data management workflow to create synthetic Whole Genome Sequence clinical data.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">generatervis</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="generatervis">generatervis<a class="anchor" aria-label="anchor" href="#generatervis"></a>
</h1></div>
<!-- badges: start -->

<!-- badges: end -->
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The <code>generatervis</code> package provides functions to create empty <code>.fastq</code> files, generate random reads, fill raw <code>.fastq</code> files with random reads, plot <code>.fastq</code> sequences, convert <code>.fastq</code> files to BAM files, convert BAM files to VCF files, and create metadata files for patient IDs.</p>
</div>
<div class="section level2">
<h2 id="id_️-installing-generatervis">⬇️ Installing <code>generatervis</code>
<a class="anchor" aria-label="anchor" href="#id_%EF%B8%8F-installing-generatervis"></a>
</h2>
<p>You can install the development version of <code>generatervis</code> from <a href="https://github.com/" class="external-link">GitHub</a> with</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("pak")  </span></span>
<span><span class="fu">pak</span><span class="fu">::</span><span class="fu"><a href="https://pak.r-lib.org/reference/pak.html" class="external-link">pak</a></span><span class="op">(</span><span class="st">"Clinical-Informatics-Collaborative/generatervis"</span><span class="op">)</span>  </span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; → Will update 1 package.</span></span>
<span><span class="co">#&gt; → The package (0 B) is cached.</span></span>
<span><span class="co">#&gt; + generatervis 0.1.0 → 0.1.0 👷🏿🔧 (GitHub: 08cd0ba)</span></span>
<span><span class="co">#&gt; ℹ No downloads are needed, 1 pkg is cached</span></span>
<span><span class="co">#&gt; ✔ Got generatervis 0.1.0 (source) (45.14 kB)</span></span>
<span><span class="co">#&gt; ℹ Packaging generatervis 0.1.0</span></span>
<span><span class="co">#&gt; ✔ Packaged generatervis 0.1.0 (375ms)</span></span>
<span><span class="co">#&gt; ℹ Building generatervis 0.1.0</span></span>
<span><span class="co">#&gt; ✔ Built generatervis 0.1.0 (586ms)</span></span>
<span><span class="co">#&gt; ✔ Installed generatervis 0.1.0 (github::Clinical-Informatics-Collaborative/generatervis@08cd0ba) (19ms)</span></span>
<span><span class="co">#&gt; ✔ 1 pkg: upd 1, dld 1 (NA B) [5.6s]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Create an empty raw <code>.fastq</code> file for the specified patient.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/create_empty_fastq.html">create_empty_fastq</a></span><span class="op">(</span>patient_id<span class="op">=</span><span class="st">"patient_123"</span><span class="op">)</span></span></code></pre></div>
<p>Generate a random sample of <code>reads</code> for a Whole Genome Sequencing (WGS) dataset for the specified patient ID.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/rreads.html">rreads</a></span><span class="op">(</span>patient_id<span class="op">=</span><span class="st">"patient_123"</span>, n <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></span></code></pre></div>
<p>Populate the <code>.fastq</code> file with the random reads.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/fill_fastq.html">fill_fastq</a></span><span class="op">(</span>patient_id<span class="op">=</span><span class="st">"patient_123"</span>,output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">2</span>, read_length <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p>(Optional) Plot the nucleotide sequences in the <code>.fastq</code> file in a grid format.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/fastq_plot.html">fastq_plot</a></span><span class="op">(</span>patient_id<span class="op">=</span><span class="st">"patient_123"</span>, output_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">2</span>, read_length <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code></pre></div>
<p>Convert the (raw) <code>.fastq</code> file to a (processed) <code>.bam</code> file using a dummy <code>.sam</code> format.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">patient_id</span> <span class="op">&lt;-</span> <span class="st">"patient_123"</span></span>
<span><span class="va">fastq_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">output_dir</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">patient_id</span>, <span class="st">".fastq"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/fill_fastq.html">fill_fastq</a></span><span class="op">(</span><span class="va">patient_id</span>, <span class="va">output_dir</span>, n <span class="op">=</span><span class="fl">2</span>, read_length<span class="op">=</span><span class="fl">8</span><span class="op">)</span></span>
<span><span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/fastq_to_bam.html">fastq_to_bam</a></span><span class="op">(</span><span class="va">fastq_file</span>, <span class="va">patient_id</span>, <span class="va">output_dir</span>, sam_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">output_dir</span>, <span class="st">"/"</span>, <span class="va">patient_id</span>, <span class="st">".sam"</span><span class="op">)</span>, reference <span class="op">=</span> <span class="st">"chr1"</span><span class="op">)</span></span></code></pre></div>
<p>To create the corresponding <code>.bam</code> file, use the <code>samtools</code> command-line tool.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="co"># samtools view -Sb path_to/file_name.sam &gt; path_to/file_name.bam</span></span></code></pre></div>
<p>Convert the (processed) <code>.bam</code> file to a (summarised) <code>.vcf</code> file format.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>patient_id <span class="ot">&lt;-</span> “patient_123”</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>generatervis<span class="sc">::</span><span class="fu">bam_to_vcf</span>(patient_id, <span class="at">output_dir =</span> <span class="fu">tempdir</span>(), <span class="at">vcf_file =</span> <span class="fu">paste0</span>(output_dir, <span class="st">"/"</span>, patient_id, <span class="st">".vcf"</span>))</span></code></pre></div>
<p>Create the metadata files to upload to <a href="https://github.com/Clinical-Informatics-Collaborative/data_storage_portal" class="external-link"><code>data_storage_repository</code></a></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">patient_id</span> <span class="op">&lt;-</span> <span class="st">"patient_123"</span></span>
<span><span class="va">output_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">generatervis</span><span class="fu">::</span><span class="fu"><a href="reference/create_metadata.html">create_metadata</a></span><span class="op">(</span><span class="va">patient_id</span>, <span class="va">output_dir</span><span class="op">)</span></span></code></pre></div>
<p>When these metadata <code>.txt</code> files are ready, they can be uploaded to <a href="https://github.com/Clinical-Informatics-Collaborative/data_storage_portal" class="external-link"><code>data_storarge_repository</code></a> by forking the repository and creating a pull request.</p>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<p>You can find detailed documentation and tutorials at the package website: <a href="https://clinical-informatics-collaborative.github.io/generatervis/" class="uri">https://clinical-informatics-collaborative.github.io/generatervis/</a></p>
<ul>
<li>
<strong><a href="https://clinical-informatics-collaborative.github.io/generatervis/">Reference manual</a></strong>: Full list of functions with detailed descriptions.</li>
</ul>
<p>For in-session help:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View documentation for a specific function</span></span>
<span><span class="op">?</span><span class="va">create_empty_fastq</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<p>To be added.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>To be added.</p>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<p>To be added.</p>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>To be added.</p>
</div>
<div class="section level2">
<h2 id="acknowledgments">Acknowledgments<a class="anchor" aria-label="anchor" href="#acknowledgments"></a>
</h2>
<p>This package is written as a part of the volunteer programme by the <a href="https://wehi-researchcomputing.github.io/" class="external-link">Research Computing Program, Walter and Eliza Hall Institute of Medical Research</a>, mentored by <a href="https://github.com/rowlandm" class="external-link">Rowland Mosbergen</a></p>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing generatervis</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jyoti Bhogal <br><small class="roles"> Maintainer </small>   </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml" class="external-link"><img src="https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/Clinical-Informatics-Collaborative/generatervis" class="external-link"><img src="https://codecov.io/gh/Clinical-Informatics-Collaborative/generatervis/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jyoti Bhogal.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
