---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# generatervis

<!-- badges: start -->

[![R-CMD-check](https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml) [![Codecov test coverage](https://codecov.io/gh/Clinical-Informatics-Collaborative/generatervis/graph/badge.svg)](https://app.codecov.io/gh/Clinical-Informatics-Collaborative/generatervis)

<!-- badges: end -->

The goal of `generatervis` is to generate and visualise Clinical data.

## Installation

You can install the development version of `generatervis` from [GitHub](https://github.com/) with:

``` {r}
# install.packages("pak")
pak::pak("Clinical-Informatics-Collaborative/generatervis")
```

## Example

# To create an empty raw (.fastq) file, use the following steps:

1. Define the patient ID for which you want to create an empty FASTQ file. 

2.1 When output directory is not specified, the `.fastq` file will be saved in the current working directory.

```{r example1}
# Create an empty FASTQ file for patient ID "patient_123"
  generatervis::create_empty_fastq("patient_123")
```

2.2 When output directory is specified, the `.fastq` file will be saved in the specified directory.
```{r example2}
# Create an empty FASTQ file for patient ID "patient_456" in a specific directory
  generatervis::create_empty_fastq("patient_456", output_dir = ".")
```

# To generate a random sample of `reads` for a Whole Genome Sequencing (WGS) dataset, use the following steps:

1. Define the patient ID for which you want to create an empty FASTQ file. 
2. Specify the number of reads you want to generate.
3. Optionally, specify the read length (default is 8). It should be multiple of 4.
4. The generated reads will be saved in a `.fastq` file in the current working directory.

Example 1: When read length is not specified, it will default to 8.
```{r example3}
# Generate 5 random reads for patient ID "patient_123"
 generatervis::rreads("patient_123", n = 5)
```

Example 2: When read length is specified, it will be set to the specified value.
```{r example4}
# Generate 5 random reads for patient ID "patient_456" with a read length of 12
 generatervis::rreads("patient_456", n = 5, read_length = 12)
```

# To fill a FASTQ file with random reads

1. Define the patient ID for which you want to fill the FASTQ file.
2. Specify the number of reads you want to generate.
3. Optionally, specify the read length (default is 8). It should be multiple of 4.
4. The generated reads will be saved in a `.fastq` file in the current working directory by default. If another directory is specified, the file will be saved there.

Example: When read length is not specified, it will default to 8.
```{r example5}
# Fill a FASTQ file for patient ID "patient_123" with 2 reads of length 8
 generatervis::fill_fastq("patient_123",output_dir = ".", n = 2, read_length = 8)
```

