---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# generatervis

<!-- badges: start -->

[![R-CMD-check](https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml) [![Codecov test coverage](https://codecov.io/gh/Clinical-Informatics-Collaborative/generatervis/graph/badge.svg)](https://app.codecov.io/gh/Clinical-Informatics-Collaborative/generatervis)

<!-- badges: end -->

The goal of `generatervis` is to generate and visualise Clinical data.

## ‚¨áÔ∏è Installing `generatervis`

You can install the development version of `generatervis` from [GitHub](https://github.com/) with

```{r, echo = TRUE}  
# install.packages("pak")  
pak::pak("Clinical-Informatics-Collaborative/generatervis")  
```  

## Demo of the workflow

The `generatervis` package provides functions to create empty `.fastq` files, generate random reads, fill raw `.fastq` files with random reads, plot `.fastq` sequences, convert `.fastq` files to BAM files, convert BAM files to VCF files, and create metadata files for patient IDs.


### üßë‚Äçüíª Creating an empty raw `.fastq` file

* Define the patient ID for which you want to create an empty raw `.fastq` file. 
  * When output directory is not specified, the `.fastq` file will be saved in the current working directory.

```{r example1}
# Create an empty `.fastq` file for patient ID "patient_123"
  generatervis::create_empty_fastq("patient_123")
```

  * When output directory is specified, the `.fastq` file will be saved in the specified directory.
```{r example2}
# Create an empty `.fastq` file for patient ID "patient_456" in a specific directory
  generatervis::create_empty_fastq("patient_456", output_dir = ".")
```

### üßë‚Äçüíª Generating a random sample of `reads` for a Whole Genome Sequencing (WGS) dataset

To create an empty random reads, define the patient ID, specify the number of reads to generate, and optionally set the read length (default is 8, and it must be a multiple of 4). The resulting reads will be saved in the current working directory.

See the below examples:

```{r example3}
# Generate 5 random reads for patient ID "patient_123"
 generatervis::rreads("patient_123", n = 5)
```

```{r example4}
# Generate 5 random reads for patient ID "patient_456" with a read length of 12
  set.seed(1067)
 generatervis::rreads("patient_456", n = 5, read_length = 12)
```

### ‚ûïÔ∏è Filling a raw `.fastq` file with random reads

To fill a raw `.fastq` file, define the patient ID, specify the number of reads to generate, and optionally set the read length (default is 8 and must be a multiple of 4). The generated reads will be saved in a `.fastq` file in the current working directory by default, or in a specified directory if provided.

```{r example5}
# Fill a raw `.fastq` file for patient ID "patient_123" with 2 reads of length 8
 generatervis::fill_fastq("patient_123",output_dir = ".", n = 2, read_length = 8)
```

### üìä Visualising raw `.fastq` data by plotting nucleotide sequences in a grid format

To plot nucleotide sequences as a heatmap, define the patient ID and specify the number of reads to include in the plot; you may also optionally set the read length (default is 8 and must be a multiple of 4). The resulting plot will be saved in the current working directory by default, or in a specified directory if provided.

```{r example6}
# Plot nucleotide sequences for patient ID "patient_123" with 2 reads of length 8
 generatervis::fastq_plot("patient_123", output_dir = ".", n = 2, read_length = 8)
```


### ‚öôÔ∏è Converting a raw `.fastq` file to a processed `.bam` file using a dummy `.sam` format

To convert a raw `.fastq` file to a processed `.bam` file, define the patient ID and specify the number of reads to include in the plot; you may also optionally set the read length (default is 8 and must be a multiple of 4). This will create a dummy `.sam` file. The `.sam` file will contain the same number of reads as the `.fastq` file, and the sequence and quality scores will be identical.

```{r example7}
# Convert a `.fastq` file to a `.bam` file
output_dir <- tempdir()
n <- 2
read_length <- 8
patient_id <- "test_patient"
fastq_file <- file.path(output_dir, paste0(patient_id, ".fastq"))
generatervis::fill_fastq(patient_id, output_dir, n, read_length)
generatervis::fastq_to_bam(fastq_file, sam_file = paste0(patient_id,".sam"), reference = "chr1")
```

To create the corresponding `.bam` file, use the `samtools` command-line tool. The `.bam` file will be created in the same directory as the `.sam` file. For this, run the following code in the command line interface:

```{bash, example8}
# samtools view -Sb dummy.sam > dummy.bam
```

### ‚öôÔ∏è Converting a processed `.bam` file to a summarised `.vcf` file format

To convert a `.bam` file, define the patient ID and specify the desired output `.vcf` file name.

```{r, example9}
patient_id <- "patient_123"
generatervis::bam_to_vcf(patient_id = patient_id, vcf_file ="output.vcf")
```

### ‚¨ÜÔ∏è Creating the metadata files to upload to [`data_storage_repository`](https://github.com/Clinical-Informatics-Collaborative/data_storage_portal)

```{r, example10}
patient_id <- "patient_123"
tmp_dir <- tempdir()
metadata <- generatervis::create_metadata(patient_id, output_dir = tmp_dir)
```

The metadata files will be written in the current working directory.

When these metadata `.txt` files are ready, they can be uploaded to [`data_storarge_repository`](https://github.com/Clinical-Informatics-Collaborative/data_storage_portal) by forking the repository and creating a pull request.
