---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# generatervis

<!-- badges: start -->

[![R-CMD-check](https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/Clinical-Informatics-Collaborative/generatervis/actions/workflows/R-CMD-check.yaml) [![Codecov test coverage](https://codecov.io/gh/Clinical-Informatics-Collaborative/generatervis/graph/badge.svg)](https://app.codecov.io/gh/Clinical-Informatics-Collaborative/generatervis)

<!-- badges: end -->

The goal of `generatervis` is to generate and visualise Clinical data.

##‚¨áÔ∏è Installing

You can install the development version of `generatervis` from [GitHub](https://github.com/) with

```{r, echo = TRUE}  
# install.packages("pak")  
pak::pak("Clinical-Informatics-Collaborative/generatervis")  
```  

## Demo of the workflow

The `generatervis` package provides functions to create empty `.fastq` files, generate random reads, fill raw `.fastq` files with random reads, plot `.fastq` sequences, convert `.fastq` files to BAM files, convert BAM files to VCF files, and create metadata files for patient IDs.


### üßë‚Äçüíª Creating an empty raw `.fastq` file

* Define the patient ID for which you want to create an empty raw `.fastq` file. 
  * When output directory is not specified, the `.fastq` file will be saved in the current working directory.

```{r example1}
# Create an empty `.fastq` file for patient ID "patient_123"
  generatervis::create_empty_fastq("patient_123")
```

  * When output directory is specified, the `.fastq` file will be saved in the specified directory.
```{r example2}
# Create an empty `.fastq` file for patient ID "patient_456" in a specific directory
  generatervis::create_empty_fastq("patient_456", output_dir = ".")
```

### üßë‚Äçüíª Generating a random sample of `reads` for a Whole Genome Sequencing (WGS) dataset

* Define the patient ID for which you want to create an empty raw `.fastq` file. 
* Specify the number of reads you want to generate.
* Optionally, specify the read length (default is 8). It should be multiple of 4.
* The generated reads will be saved in a `.fastq` file in the current working directory.

When read length is not specified, it will default to 8.
```{r example3}
# Generate 5 random reads for patient ID "patient_123"
 generatervis::rreads("patient_123", n = 5)
```

When read length is specified, it will be set to the specified value.
```{r example4}
# Generate 5 random reads for patient ID "patient_456" with a read length of 12
  set.seed(1067)
 generatervis::rreads("patient_456", n = 5, read_length = 12)
```

### ‚ûïÔ∏è Filling a raw `.fastq` file with random reads

* Define the patient ID for which you want to fill the raw `.fastq` file.
* Specify the number of reads you want to generate.
* Optionally, specify the read length (default is 8). It should be multiple of 4.
* The generated reads will be saved in a `.fastq` file in the current working directory by default. If another directory is specified, the file will be saved there.

When read length is not specified, it will default to 8.
```{r example5}
# Fill a raw `.fastq` file for patient ID "patient_123" with 2 reads of length 8
 generatervis::fill_fastq("patient_123",output_dir = ".", n = 2, read_length = 8)
```

### üìä Visualising raw `.fastq` data by plotting nucleotide sequences in a grid format

* Define the patient ID for which you want to plot the nucleotide sequences.
* Specify the number of reads you want to plot.
* Optionally, specify the read length (default is 8). It should be multiple of 4.
* The generated plot will be saved in the current working directory by default. If another directory is specified, the plot will be saved there.

```{r example6}
# Plot nucleotide sequences for patient ID "patient_123" with 2 reads of length 8
 generatervis::fastq_plot("patient_123", output_dir = ".", n = 2, read_length = 8)
```


### ‚öôÔ∏è Converting a raw `.fastq` file to a processed `.bam` file using a dummy `.sam` format

```{r example7}
# Convert a `.fastq` file to a `.bam` file
output_dir <- tempdir()
n <- 2
read_length <- 8
patient_id <- "test_patient"
fastq_file <- file.path(output_dir, paste0(patient_id, ".fastq"))
generatervis::fill_fastq(patient_id, output_dir, n, read_length)
generatervis::fastq_to_bam(fastq_file, sam_file = paste0(patient_id,".sam"), reference = "chr1")
```

This will create a `.sam` file in the specified output directory. The `.sam` file will contain the same number of reads as the `.fastq` file, and the sequence and quality scores will be identical.

To create the corresponding `.bam` file, you can use the `samtools` command-line tool. The `.bam` file will be created in the same directory as the `.sam` file. For this, run the following code in the command line interface:

```{bash, example8}
# samtools view -Sb dummy.sam > dummy.bam
```

### ‚öôÔ∏è Converting a processed `.bam` file to a summarised `.vcf` file format

* Define the patient ID for which you want to convert the `.bam` file.
* Specify the output `.vcf` file name.

```{r, example9}
patient_id <- "patient_123"
generatervis::bam_to_vcf(patient_id = patient_id, vcf_file ="output.vcf")
```

The metadata files will be written in the current working directory.

When these metadata `.txt` files are ready, they can be uploaded to [`data_storarge_repository`](https://github.com/Clinical-Informatics-Collaborative/data_storage_portal) by forking the repository and creating a pull request.
